src/diffusers/loaders/lora_pipeline.py

```diff
@@ -2113,8 +2264,124 @@ def unfuse_lora(self, components: List[str] = ["transformer", "text_encoder"], *
+        prefix = prefix or cls.transformer_name
+        for key in list(state_dict.keys()):
+            if key.split(".")[0] == prefix:
+                state_dict[key.replace(f"{prefix}.", "")] = state_dict.pop(key)
+
+        def get_submodule(module, name):
+            for part in name.split("."):
+                if len(name) == 0:
+                    break
+                if not hasattr(module, part):
+                    raise AttributeError(f"Submodule '{part}' not found in '{module}'.")
+                module = getattr(module, part)
+            return module
+
+        # Expand transformer parameter shapes if they don't match lora
+        has_param_with_shape_update = False
+
+        for name, module in transformer.named_modules():
+            if isinstance(module, torch.nn.Linear):
+                module_weight = module.weight.data
+                module_bias = module.bias.data if hasattr(module, "bias") else None
+                bias = module_bias is not None
+                name_split = name.split(".")
+
+                lora_A_name = f"{name}.lora_A"
+                lora_B_name = f"{name}.lora_B"
+                lora_A_weight_name = f"{lora_A_name}.weight"
+                lora_B_weight_name = f"{lora_B_name}.weight"
+
+                if lora_A_weight_name not in state_dict.keys():
+                    continue
+
+                in_features = state_dict[lora_A_weight_name].shape[1]
+                out_features = state_dict[lora_B_weight_name].shape[0]
+
+                if tuple(module_weight.shape) == (out_features, in_features):
+                    continue
+
+                module_out_features, module_in_features = module_weight.shape
+                if out_features < module_out_features or in_features < module_in_features:
+                    raise NotImplementedError(
+                        f"Only LoRAs with input/output features higher than the current module's input/output features "
+                        f"are currently supported. The provided LoRA contains {in_features=} and {out_features=}, which "
+                        f"are lower than {module_in_features=} and {module_out_features=}. If you require support for "
+                        f"this please open an issue at https://github.com/huggingface/diffusers/issues."
+                    )
+
+                logger.debug(
+                    f'Expanding the nn.Linear input/output features for module="{name}" because the provided LoRA '
+                    f"checkpoint contains higher number of features than expected. The number of input_features will be "
+                    f"expanded from {module_in_features} to {in_features}, and the number of output features will be "
+                    f"expanded from {module_out_features} to {out_features}."
+                )
+
+                has_param_with_shape_update = True
+                parent_module_name = ".".join(name_split[:-1])
+                current_module_name = name_split[-1]
```

Comment:
```suggestion
                parent_module_name, _, current_module_name = name.rpartition(".")
```

Reply:
Cc: @a-r-r-o-w

Reply:
I forgot this was a thing 😅 Very neat, will update!